File: Source/Magazine.Web/src/app/server-communication/events/comment/comment.service.event.ts
import { Injectable } from "@angular/core";
import { CommentService } from "src/app/comment/services/comment.service";
import { EntityServiceEvent } from "../../contracts/entity.service.event";
import { CommentEventArgument } from "./comment-event-argument";

/** Событие сервиса сущности "Комментарий". */
@Injectable({
    providedIn: 'root'
})
export abstract class CommentServiceEvent extends EntityServiceEvent {

    eventArgument: CommentEventArgument;

    constructor(className: string,
                eventArgument: CommentEventArgument) {
        super(className);
        this.eventArgument = eventArgument;
    }

    abstract visit(service: CommentService): void;
}

File: Source/Magazine.API/Contracts/Event/CommentServiceEvent.cs
using Newtonsoft.Json;

namespace Magazine.API.Contracts.Event
{
    /// <summary>
    /// Событие сервиса сущности "Комментарий".
    /// </summary>
    public abstract class CommentServiceEvent : EntityServiceEvent
    {
        [JsonProperty]
        /// <summary>
        /// Идентификатор комментария.
        /// </summary>
        protected int Id { get; set; }
    }
}
